From b95e9043b33104b1e2b8224f6a5b1493ca6757c9 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Sun, 1 Sep 2013 16:22:01 -0700
Subject: [PATCH 288/974] MIPS/EDAC: Poll for LMC_INT_REG[nxm_wr_err]

Add missing error interrupt, nxm_wr_err, write to no-existent memory.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/edac/octeon_edac-lmc.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/edac/octeon_edac-lmc.c b/drivers/edac/octeon_edac-lmc.c
index d66c3fc..8905370 100644
--- a/drivers/edac/octeon_edac-lmc.c
+++ b/drivers/edac/octeon_edac-lmc.c
@@ -82,6 +82,14 @@ static void octeon_lmc_edac_poll_o2(struct mem_ctl_info *mci)
 		int_reg.s.ded_err = -1;	/* Done, re-arm */
 		do_clear = true;
 	}
+	if (int_reg.s.nxm_wr_err) {
+		snprintf(msg, sizeof(msg), "NXM_WR_ERR: Write to non-existent memory");
+		edac_mc_handle_error(HW_EVENT_ERR_CORRECTED, mci, 1, 0, 0, 0,
+				     -1, -1, -1, msg, "");
+		int_reg.s.nxm_wr_err = -1;	/* Done, re-arm */
+		do_clear = true;
+	}
+
 	if (do_clear)
 		cvmx_write_csr(CVMX_LMCX_INT(mci->mc_idx), int_reg.u64);
 }
-- 
2.6.2

